client/cmd.go

```diff
@@ -55,3 +58,59 @@ func ValidateCmd(cmd *cobra.Command, args []string) error {
 
 	return cmd.Help()
 }
+
+// ReadGetCommandFlags returns an updated Context with fields set based on flags
+// defined in GetCommands. An error is returned if any flag query fails.
+func ReadGetCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, error) {
+	if flagSet.Changed(flags.FlagHeight) {
+		height, err := flagSet.GetInt64(flags.FlagHeight)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithHeight(height)
+	}
+
+	if flagSet.Changed(flags.FlagTrustNode) {
+		trustNode, err := flagSet.GetBool(flags.FlagTrustNode)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithTrustNode(trustNode)
+	}
+
+	if flagSet.Changed(flags.FlagUseLedger) {
+		useLedger, err := flagSet.GetBool(flags.FlagUseLedger)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithUseLedger(useLedger)
+	}
+
+	if clientCtx.Keyring == nil && flagSet.Changed(flags.FlagKeyringBackend) {
```

Comment:
@alexanderbez we should not use `Changed()` anymore. Instead, we should rely on the flag's default value.

Reply:
Why not? Also, this is **intentional**. A `client.Context` is passed into a command constructor/closure. This `client.Context` may already have fields set (e.g. `Client`), so the use of `Changed` states only _overwrite_ the field if it has been explicitly set. Otherwise, bugs can be introduced where fields are wiped out.

Reply:
So upon further thought, this really only applies to certain fields/flags. e.g. there is no point in asking if `FlagHeight` changed, we should just read it. However, we should keep `Changed` for other situations like `Client` and `Keyring`.